search/search_repository_whitebox_test.go

```diff
@@ -45,189 +42,209 @@ func (s *searchRepositoryWhiteboxTest) SetupTest() {
-			optionalKeywords := []string{workItemURLInSearchString, strconv.Itoa(createdWorkItem.Number)}
-
-			// We will now check the legitimacy of the search results.
-			// Iterate through all search results and see whether they meet the criteria
-
-			for _, workItemValue := range workItemList {
-				s.T().Log("Found search result  ", workItemValue.ID)
-
-				for _, keyWord := range allKeywords {
-
-					workItemTitle := ""
-					if workItemValue.Fields[workitem.SystemTitle] != nil {
-						workItemTitle = strings.ToLower(workItemValue.Fields[workitem.SystemTitle].(string))
-					}
-					workItemDescription := ""
-					if workItemValue.Fields[workitem.SystemDescription] != nil {
-						descriptionField := workItemValue.Fields[workitem.SystemDescription].(rendering.MarkupContent)
-						workItemDescription = strings.ToLower(descriptionField.Content)
-					}
-					workItemNumber := 0
-					if workItemValue.Fields[workitem.SystemNumber] != nil {
-						workItemNumber = workItemValue.Fields[workitem.SystemNumber].(int)
-					}
-					keyWord = strings.ToLower(keyWord)
-
-					if strings.Contains(workItemTitle, keyWord) || strings.Contains(workItemDescription, keyWord) {
-						// Check if the search keyword is present as text in the title/description
-						s.T().Logf("Found keyword %s in workitem %s", keyWord, workItemValue.ID)
-					} else if stringInSlice(keyWord, optionalKeywords) && strings.Contains(keyWord, strconv.Itoa(workItemValue.Number)) {
-						// If not present in title/description then it should be a URL or ID
-						s.T().Logf("Found keyword '%s' as number '%s' from the URL", keyWord, strconv.Itoa(workItemValue.Number))
-					} else {
-						s.T().Errorf("'%s' neither found in title '%s' nor in the description: '%s' for workitem number %d", keyWord, workItemTitle, workItemDescription, workItemNumber)
-					}
+		// We will now check the legitimacy of the search results.
+		// Iterate through all search results and see whether they meet the criteria
+		for _, workItemValue := range workItemList {
+			s.T().Logf("Examining workitem id=`%v` number=`%d` using keywords %v", workItemValue.ID, workItemValue.Number, allKeywords)
+			for _, keyWord := range allKeywords {
+				keyWord = strings.ToLower(keyWord)
+				s.T().Logf("Verifying workitem id=`%v` number=`%d` for keyword `%s`...", workItemValue.ID, workItemValue.Number, keyWord)
+				workItemTitle := ""
+				if workItemValue.Fields[workitem.SystemTitle] != nil {
+					workItemTitle = strings.ToLower(workItemValue.Fields[workitem.SystemTitle].(string))
+				}
+				workItemDescription := ""
+				if workItemValue.Fields[workitem.SystemDescription] != nil {
+					descriptionField := workItemValue.Fields[workitem.SystemDescription].(rendering.MarkupContent)
+					workItemDescription = strings.ToLower(descriptionField.Content)
 				}
+				assert.True(s.T(),
+					strings.Contains(workItemTitle, keyWord) || strings.Contains(workItemDescription, keyWord),
```

Comment:
okay! a question
what would happen when user types in just `10` to search. Will it return a WI with id=10? Can we have a test for that too?
Currently, we are using this `search` when a user wants to link WIs.

Reply:
@pranavgore09 I added a test for searching by WI _number_, but not by _id_ . Also, I used the `number:` prefix, but maybe I should also add a test to search without the prefix ? My question is: `ID` is supposed to be an internal UUID, so do we expect users to use it to look-up a work item ?

Reply:
oh! yeah, I was concerned with UUID in the search bar, but yes we may not need that case.
Generally, `number` will be used. Not sure if UI need to make this change or they are already using `number` only. I will confirm with the team and respond here.
thanks @xcoulon for an explanation.
And test without prefix would be helpful IMHO

Reply:
Yeah on UI we are using `number` while search and not `uuid`!